pkg/v1/google/auth.go

```diff
@@ -115,6 +121,13 @@ func (gs gcloudSource) Token() (*oauth2.Token, error) {
 	// Don't attempt to interpret stderr, just pass it through.
 	cmd.Stderr = os.Stderr
 
+	if _, err := exec.LookPath("gcloud"); err != nil {
```

Comment:
Can we check this earlier so we don't have to communicate with callers via `anonError`?

I'm thinking at the top of `NewGcloudAuthenticator` we do something like:
```go
cmd := GetGcloudCmd()
if _, err := exec.LookPath(cmd.Path); err != nil {
  // TODO(#390): Use better logger.
  // gcloud is not available, fall back to anonymous
  log.Println("gcloud binary not found")
  return authn.Anonymous, nil
}
ts := gcloudSource{cmd}
...
```

Reply:
Fair, I’ll update ✌🏼